contracts/modules/v2/Assessment.sol

```diff
@@ -70,34 +70,41 @@ contract Assessment is IAssessment, MasterAwareV2 {
     return assessments[assessmentId].poll;
   }
 
-  function getRewards(address user) external view returns (
-    uint total,
-    uint withdrawable,
+  function getRewards(address user) external override view returns (
+    uint totalPendingAmount,
+    uint withdrawableAmount,
     uint withdrawableUntilIndex
   ) {
-    uint104 rewardsWithdrawnUntilIndex = stakeOf[user].rewardsWithdrawnUntilIndex;
+    uint104 rewardsWithdrawableFromIndex = stakeOf[user].rewardsWithdrawableFromIndex;
     Vote memory vote;
     Assessment memory assessment;
     uint voteCount = votesOf[user].length;
-    for (uint i = rewardsWithdrawnUntilIndex; i < voteCount; i++) {
+    bool hasReachedUnwithdrawableReward = false;
+
+    for (uint i = rewardsWithdrawableFromIndex; i < voteCount; i++) {
       vote = votesOf[user][i];
       assessment = assessments[vote.assessmentId];
 
-      // If withdrawableUntilIndex has been assigned before, continue calculating the total accrued
+      // If hasReachedUnwithdrawableReward is true, skip and continue calculating the pending total
       // rewards.
       if (
-        withdrawableUntilIndex == 0 &&
+        !hasReachedUnwithdrawableReward &&
```

Comment:
why does the condition here change?

before and after withdrawableUntilIndex = i so it would only enter the if when the index is 0, can the index be and you'd want to enter the if?

Reply:
The scenario is one withdrawable reward on only one vote (index 0). withdrawableUntilIndex should return voteCount in that case (1). A second bug here was that for two unwithdrawable rewards, it would not stop at index 0 and would have considered vote 0 withdrawable even though it wasn't.